### DISCUSSION TITLE
Adding a Service

### DISCUSSION THREAD

**Discussion by timberlakeis** [Security-Onion-Solutions/securityonion#2009] [Unsupported Versions] [answered]
Today I wanted to test out SecOnion 2.3 and I went with the Standalone option in my homelab. 

When I initially ran through the install I only elected to have a few services install at that time. What is the preferred method for adding one of the services on after going through the setup script?

Essentially, I skipped Grafana on initial install and now I want to add it.

Thanks for the help.

**Comment by weslambert** [comment, marked as answer]
Typically, you can flip the switch from 0 to 1, or False to True in global.sls or minion.sls, then perform a salt-call state.apply <service>, however.  There may have been configuration that happened during setup that provides necessary information for the service to be configured correctly.

**Reply by timberlakeis** [reply]
Thanks! This makes sense, but unfortunately I receive this error when I attempt to apply.

![image](https://user-images.githubusercontent.com/65873661/99822608-09970100-2b19-11eb-8655-e9409c158faa.png)

**Reply by weslambert** [reply]
Sorry, looks like GH stripped the `<service>` I had in tags.  It would be `state.apply <service>`, or `so-<service>-start`

**Reply by weslambert** [reply]
And you can add a `queue=True` to that or run the `so-<service>-start --force` to kill current running jobs and put your job in the front of the queue.

**Reply by timberlakeis** [reply]
Perfect! Thanks for your quick and helpful replies!
